ESM types are described incorrectly

https://arethetypeswrong.github.io/?p=preact-render-to-string%406.6.5

I started working on this a bit, but the repo needs a bit of a larger overhaul that would be better for a maintainer to do.

```jsonc
// package.json
{
  "dependencies": {
    "preact": "^10.28.2",
    "preact-render-to-string": "^6.6.5",
    "typescript": "^5.9.3"
  }
}
```

```jsonc
// tsconfig.json
{
  "compilerOptions": {
    "module": "nodenext",
    "target": "esnext",
    "types": [],
    "noEmit": true,
    "noUncheckedIndexedAccess": true,
    "exactOptionalPropertyTypes": true,
    "strict": true,
    "jsx": "react-jsx",
    "verbatimModuleSyntax": true,
    "isolatedModules": true,
    "noUncheckedSideEffectImports": true,
    "moduleDetection": "force",
    "skipLibCheck": true,
  }
}
```

```ts
// main.mts
import renderToString, { shallowRender, render } from "preact-render-to-string/jsx";
import { h } from "preact";

console.log(shallowRender);
console.log(render);
console.log(renderToString(h("div", {})))
```

```
> npx tsc --noEmit
main.mts:6:13 - error TS2349: This expression is not callable.
  Type 'typeof import("V:/scratch/node_modules/preact-render-to-string/dist/jsx")' has no call signatures.

6 console.log(renderToString(h("div", {})))
              ~~~~~~~~~~~~~~


Found 1 error in main.mts:6
```

--- a/.changeset/gentle-eyes-trade.md
+++ b/.changeset/gentle-eyes-trade.md
@@ -1,0 +1,5 @@
+---
+"preact-render-to-string": patch
+---
+
+fix: use cjs types instead of esm for jsx export to fix default export from esm

--- a/jsx.d.ts
+++ b/jsx.d.ts
@@ -10,15 +10,24 @@ interface Options {
 	skipFalseAttributes?: boolean;
 }
 
-export default function renderToStringPretty(
+declare function renderToStringPretty(
 	vnode: VNode,
 	context?: any,
 	options?: Options
 ): string;
-export function render(vnode: VNode, context?: any, options?: Options): string;
 
-export function shallowRender(
-	vnode: VNode,
-	context?: any,
-	options?: Options
-): string;
+declare namespace renderToStringPretty {
+	export function render(
+		vnode: VNode,
+		context?: any,
+		options?: Options
+	): string;
+
+	export function shallowRender(
+		vnode: VNode,
+		context?: any,
+		options?: Options
+	): string;
+}
+
+export = renderToStringPretty;

--- a/src/jsx.d.ts
+++ b/src/jsx.d.ts
@@ -10,15 +10,24 @@ interface Options {
 	skipFalseAttributes?: boolean;
 }
 
-export default function renderToStringPretty(
+declare function renderToStringPretty(
 	vnode: VNode,
 	context?: any,
 	options?: Options
 ): string;
-export function render(vnode: VNode, context?: any, options?: Options): string;
 
-export function shallowRender(
-	vnode: VNode,
-	context?: any,
-	options?: Options
-): string;
+declare namespace renderToStringPretty {
+	export function render(
+		vnode: VNode,
+		context?: any,
+		options?: Options
+	): string;
+
+	export function shallowRender(
+		vnode: VNode,
+		context?: any,
+		options?: Options
+	): string;
+}
+
+export = renderToStringPretty;

